src/core/lib/surface/alarm.c

```diff
@@ -93,40 +95,45 @@ static void alarm_cb(grpc_exec_ctx *exec_ctx, void *arg, grpc_error *error) {
                  (void *)alarm, &alarm->completion);
 }
 
-grpc_alarm *grpc_alarm_create(grpc_completion_queue *cq, gpr_timespec deadline,
-                              void *tag) {
+grpc_alarm *grpc_alarm_create(void *reserved) {
   grpc_alarm *alarm = gpr_malloc(sizeof(grpc_alarm));
-  grpc_exec_ctx exec_ctx = GRPC_EXEC_CTX_INIT;
-
-  gpr_ref_init(&alarm->refs, 1);
 
 #ifndef NDEBUG
   if (GRPC_TRACER_ON(grpc_trace_alarm_refcount)) {
     gpr_log(GPR_DEBUG, "Alarm:%p created (ref: 1)", alarm);
   }
 #endif
 
+  gpr_ref_init(&alarm->refs, 1);
+  alarm->alarm.pending = false;
```

Comment:
Setting `alarm.pending` to `false` is not needed. The `grpc_timer` isn't even visible to timer threads until you do `grpc_timer_init`. I suggest removing this.

Reply:
Right, this is there to deal with the case that someone calls alarm_destroy after create without ever having set it. But the code above is the wrong way to do it because I shouldn't be poking around in the timer's internals except under the timer's shard_mu, come to think of it. So I'll fix this up.

Reply:
Actually, let me back out of my previous comment. Although it's ugly for me to poke around in the internals of timer here (and I should clean that up for style reasons), it doesn't need to be under a lock since this is the creation of the alarm struct.